src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -116,15 +140,23 @@ public void storeMessage(Message message) {
       Message message = new Message(id, sender, text, timestamp, receiver);
       messages.add(message);
     }
-
     return messages;
-  }
-
-  /** Returns the total number of messages for all users. */
-  public int getTotalMessageCount() {
-    Query query = new Query("Message");
-    PreparedQuery results = datastore.prepare(query);
-    return results.countEntities(FetchOptions.Builder.withDefaults());
-  }
+ }
+
+ private boolean hasUserRestriction(String loggedInUser, String otherUser) {
+   if (!Strings.isNullOrEmpty(loggedInUser) && !Strings.isNullOrEmpty(otherUser)) {
```

Comment:
The following pattern:

```
      if (expr) {
        return false;
      }
      return true;
```

Can be reduced to:

```
    return !expr;
```

That being said, it looks like you've reversed `false` and `true` (so it should be reduced to `return expr`).

Simplify to:?

```
private boolean hasUserRestriction(String loggedInUser, String otherUser) {
   return !Strings.isNullOrEmpty(loggedInUser) && !Strings.isNullOrempty(otherUser);
}
```

Reply:
This makes sense and makes the code cleaner. Perfect!